lib/dal/src/action/prototype.rs

```diff
@@ -362,10 +451,56 @@ impl ActionPrototype {
         Ok(action_prototype_ids)
     }
 
+    pub async fn parentage(
+        ctx: &DalContext,
+        id: ActionPrototypeId,
+    ) -> ActionPrototypeResult<ActionPrototypeParent> {
+        if let Some(schema_ids) = Self::schema_ids(ctx, id).await? {
```

Comment:
Should this error if an ActionPrototype has both Schemas and a SchemaVariant as parents?

Reply:
you know i don't have a really good answer here. It's not actually a problem if the same prototype has multiple "parents", one a variant, the other a schema. We only use parentage in authoring and "binding". But we also don't expose the ability to take an existing prototype and connect it to something else (like another schema variant, or two schemas) yet.

Reply:
Yeah, it's a nit for sure, the system shouldn't be doing this. I'd at least add a comment in parentage() though, especially since `AttributePrototypeParent::Schemas` implies that there can be multiple parents for a thing.